library/Core.cpp

```diff
@@ -948,7 +948,7 @@ command_result Core::runCommand(color_ostream &con, const std::string &first_, v
                 << "Context may be used to limit the scope of the binding, by" << endl
                 << "requiring the current context to have a certain prefix." << endl
                 << "Current UI context is: "
-                << Gui::getFocusString(Core::getTopViewscreen()) << endl;
+                << Gui::getFocusString(Core::getTopViewscreen())[0] << endl;
```

Comment:
Would it make sense to print the entire vector here?

Also, is it possible for this vector to be empty? A brief look at `Gui.cpp` suggests that it could be.

Reply:
Yeah this is janky atm. I have a fallback at the end of the focus string handler for dwarfmode that ensures at least "dwarfmode" is in the vector if it didn't add anything else. Assuming the dwarfmode screen is always present now, it should at least have that, but yeah I definitely need to update any code that relies on `getFocusString` to not just take the first element.

Reply:
the dwarfmode screen will not be in every context. only specifically when the fort map is visible, and sometimes not even then if a separate viewscreen (not just a panel) is partially covering it. For a simple example, when you first start the game and you see the dwarf face being constructed line by line, that's viewscreen_setupdwarfgamest, and the title screen (with the splash image in the background) is viewscreen_titlest

Reply:
Hah, yeah I actually found this out by hitting the launcher hotkey earlier while in the title screen and immediately thought of this comment 🙈